starter-code/src/functions-and-arrays.js

```diff
@@ -1,93 +1,432 @@
 // Iteration #1: Find the maximum
+function maxOfTwoNumbers(firstNumber, secondNumber) {
+  if (firstNumber > secondNumber) {
+    return firstNumber;
+  }
+  return secondNumber;
+}
+console.log(maxOfTwoNumbers(10, 5));
+console.log(maxOfTwoNumbers(30, 35));
 
 // Iteration #2: Find longest word
-const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
+const words = [
+  "mystery",
+  "brother",
+  "aviator",
+  "crocodile",
+  "pearl",
+  "orchard",
+  "crackpot"
+];
+function findLongestWord(arr) {
+  let longestWord = "";
+  if (arr.length <= 0) {
+    return null;
+  }
+  for (i = 0; i < arr.length; i++) {
+    if (longestWord.length < arr[i].length) {
+      longestWord = arr[i];
+    }
+  }
+  return longestWord;
+}
+//                                                                             How do I console.log "corocodile"?!!
 
 // Iteration #3: Calculate the sum
 
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
+function sumArray(numbers) {
+  let sumNum = 0;
+  for (let i = 0; i < numbers.length; i++) {
+    sumNum += numbers[i];
+  }
+  return sumNum;
+}
 
 // Iteration #4: Calculate the average
+
 // Level 1: Array of numbers
-const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
+const numbersArr = [2, 6, 9, 10, 7, 4, 1, 9];
+function averageNumbers(numbersArr) {
+  let sumNu = 0;
+  for (let i = 0; i < numbersArr.length; i++) {
+    sumNu += numbersArr[i];
+  }
+  return sumNu / numbersArr.length;
+
+  if (sumNu([])) {
+    return null;
+  }
+} //                                                                I couldn't make it return null with an empty array!! :(
 
 // Level 2: Array of strings
 const wordsArr = [
-  'seat',
-  'correspond',
-  'linen',
-  'motif',
-  'hole',
-  'smell',
-  'smart',
-  'chaos',
-  'fuel',
-  'palace'
+  "seat",
+  "correspond",
+  "linen",
+  "motif",
+  "hole",
+  "smell",
+  "smart",
+  "chaos",
+  "fuel",
+  "palace"
 ];
 
+function averageWordLength(wordsArr) {
+  let totalLength = 0;
+  if (wordsArr.length <= 0) {
+    return null;
+  }
+
+  for (let i = 0; i < wordsArr.length; i++) {
+    totalLength += wordsArr[i].length;
+  }
+  return totalLength / wordsArr.length;
+}
+
 // Iteration #5: Unique arrays
 const wordsUnique = [
-  'crab',
-  'poison',
-  'contagious',
-  'simple',
-  'bring',
-  'sharp',
-  'playground',
-  'poison',
-  'communion',
-  'simple',
-  'bring'
+  "crab",
+  "poison",
+  "contagious",
+  "simple",
+  "bring",
+  "sharp",
+  "playground",
+  "poison",
+  "communion",
+  "simple",
+  "bring"
 ];
+function uniquifyArray(wordsUnique) {}
 
 // Iteration #6: Find elements
 const wordsFind = [
-  'machine',
-  'subset',
-  'trouble',
-  'starting',
-  'matter',
-  'eating',
-  'truth',
-  'disobedience'
+  "machine",
+  "subset",
+  "trouble",
+  "starting",
+  "matter",
+  "eating",
+  "truth",
+  "disobedience"
 ];
-
+function doesWordExist([wordsFind, wordFinder]) {
+  let wordFinder = "";
```

Comment:
try not to create a let variable called the same as the parameter inside the function, if you do it, it will give you an error.

Reply:
what I mean by this, if you have `function (banana){
let banana}`
you will have an error.